compiler/test/fail_compilation/test3543.d

```diff
@@ -0,0 +1,57 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/test3543.d(135): Error: incompatible types for `(a) : (b)`: `test3543.teste.A` and `test3543.teste.B`
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=3543
+
+#line 100
```

Comment:
I do that because otherwise adding a line or two means redoing all the test output.

Reply:
What about instead:
```d
// BEGIN TEST
```
And then the test checker can be made to see something like:

```d
fail_compilation/test3543.d(TL135): Error: incompatible types for `(a) : (b)`: `test3543.teste.A` and `test3543.teste.B`
```

and properly substitute `TL135` for the real line numbers during the check?

When the test fails to pass, then the output is printed with the wrong line number (according to the editor). The checker is automated it can afford to do arbitrary calculations in nanoseconds. The user is human, his time is valuable, and shouldn't be forced to do them manually.

Reply:
I actually don't even understand why we need to worry about this. Updating the expected output should be as easy as copy/pasting, no?

Reply:
What if the checker detected just line number changes (maybe via regex) and suggested how to update the text? That way, you can just copy-paste from the error log. Something like:

```
Trivial line changes detected, if this is expected, please replace the lines to test test with this text:

...
```

Another option -- just put the test expected data at the bottom of the file (so changing the expectation lines doesn't affect the test itself)

Reply:
@schveiguy this is already a solved issue for verbatim test output messages for many years now.
```
AUTO_UPDATE=1 ./run.d run_fail_compilation
```

Reply:
I'm not surprised ;)

Reply:
@WalterBright FYI: https://github.com/dlang/dmd/tree/master/compiler/test#automatically-update-the-test_output-segments